finetune/input_pipeline.py

```diff
@@ -210,12 +196,18 @@ def _format_for_pipeline(data):
         return data_
         
     def _dataset_without_targets(self, Xs, train, context=None, update_hook=None, forced_mask=None):
-        Xs_ = self._format_for_pipeline(Xs)
-        ys_ = self._format_for_pipeline(None)
-        context_ = self._format_for_pipeline(context)
-        forced_mask_ = self._format_for_pipeline(forced_mask)
+        if callable(Xs):
+            assert forced_mask is None, "Forced mask is not supported with generator inputs"
+            def Xs_gen():
+                for x in Xs():
+                    yield (x.get("X"), x.get("y"), x.get("context"), None)
```

Comment:
This does break generate_text and the cached predict that is on this branch, but i'm not sure if thats okay?